core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -584,7 +589,31 @@ public UpdateNode plan(Update node)
         builder = planAndMappings.getSubPlan();
 
         ImmutableList.Builder<Symbol> updatedColumnValuesBuilder = ImmutableList.builder();
-        orderedColumnValues.forEach(columnValue -> updatedColumnValuesBuilder.add(planAndMappings.get(columnValue)));
+        Assignments.Builder assignmentsBuilder = Assignments.builder();
+        assignmentsBuilder.putIdentities(builder.getRoot().getOutputSymbols());
+        Map<String, ColumnMetadata> columnNameToMetadata = columnsMetadata.stream()
+                .collect(toImmutableMap(ColumnMetadata::getName, identity()));
+
+        for (int i = 0; i < targetColumnNames.size(); i++) {
+            String columName = targetColumnNames.get(i);
+            ColumnMetadata columnMetadata = columnNameToMetadata.get(columName);
+            Symbol output = symbolAllocator.newSymbol(columName, columnMetadata.getType());
+            updatedColumnValuesBuilder.add(output);
+            Expression expression = ((ProjectNode) builder.getRoot()).getAssignments().get(planAndMappings.get(orderedColumnValues.get(i)));
+            if (!columnMetadata.isNullable()) {
+                expression = new CoalesceExpression(expression, new Cast(
+                        failFunction(
+                                metadata,
+                                session,
+                                INVALID_ARGUMENTS,
+                        "NULL value not allowed for NOT NULL column: " + columName),
```

Comment:
The formatting is off. This should be aligned with the rest of the arguments to `failFunction`.